turbomoleio/output/tests/test_files.py

```diff
@@ -29,60 +29,79 @@
-              ("ridft", "h2o_dftd3-bj_not_conv"), ("ridft", "nh3_rijk_xcfun_m06"),
-              ("ridft", "b28_many_irreps"), ("grad", "h2o_std"), ("rdgrad", "h2o_dftd3-bj"),
-              ("relax", "h2o_internal"), ("relax", "h2o_cartesian"), ("relax", "no_version_header"),
-              ("statpt", "h3cbr_internal"), ("statpt", "aceton_cartesian"),
-              ("escf", "Al6_columns"), ("escf", "h2o_ridft_cosmo"), ("escf", "h2o_ridft_rpat"),
-              ("egrad", "h2o_sym"), ("egrad", "h3cbr_nosym"), ("aoforce", "aceton_full"),
-              ("aoforce", "h2_numforce"), ("jobex", "h2o_dscf"), ("jobex", "no3_ridft"),
-              ("escf_only", "h2o_ridft_cosmo")]
+excluded_tests = [('relax', 'no_version_header'),
+                  ('aoforce', 'h2_numforce')]
+files_list = [
+    (tm_exec, test_name)
+    for tm_exec, exec_tests in OUTPUTS_BASENAMES.items()
+    for test_name in exec_tests
+    if (tm_exec, test_name) not in excluded_tests
+]
 
 
 @pytest.fixture(scope="function", params=files_list, ids=[os.path.join(*f) for f in files_list])
-def cls_dict_path(request, testdir):
-    ref_name = request.param[0]
-    if ref_name == "escf_only":
-        directory = "escf"
+def cls_dict_path(request, testdir, tm_version):
+    tm_exec = request.param[0]
+    test_name = request.param[1]
+    path = os.path.join(testdir, "outputs", tm_version, tm_exec, test_name)
+    output_clses = []
+    ref_dicts = []
+    fpaths = []
+    if tm_exec == "escf":
+        output_clses.append(EscfOnlyOutput)
+        ref_dicts.append(loadfn(os.path.join(path, "ref_escf_output.json"), cls=None))
+        fpaths.append(os.path.join(path, f"{tm_exec}.log"))
+        output_clses.append(EscfOutput)
+        ref_dicts.append(loadfn(os.path.join(path, "ref_output.json"), cls=None))
+        fpaths.append(os.path.join(path, f"{tm_exec}.log"))
+    elif tm_exec == "jobex":
+        output_clses.append(JobexOutput)
+        ref_dicts.append(loadfn(os.path.join(path, "ref_output.json"), cls=None))
+        fpaths.append(os.path.join(path, "job.last"))
     else:
-        directory = ref_name
-    name = request.param[1]
-    path = os.path.join(testdir, "outputs", directory, name)
+        output_clses.append(exec_to_out_obj[tm_exec])
+        ref_dicts.append(loadfn(os.path.join(path, "ref_output.json"), cls=None))
+        fpaths.append(os.path.join(path, f"{tm_exec}.log"))
 
-    if ref_name == "escf_only":
-        json_path = path + "_outfile_escf_only.json"
-    else:
-        json_path = path + "_outfile.json"
-
-    with open(json_path) as f:
-        d = json.load(f)
-
-    exec_dict = dict(exec_to_out_obj)
-    exec_dict["jobex"] = JobexOutput
-    exec_dict["escf_only"] = EscfOnlyOutput
-
-    file_path = path + ".log" if ref_name != "jobex" else path + "_job.last"
-
-    return exec_dict[ref_name], d, file_path
+    return output_clses, ref_dicts, fpaths
 
 
 class TestParser:
 
+    @pytest.mark.parametrize("tm_version", TM_VERSIONS)
```

Comment:
So you would put tm_version as a fixture in conftest ? And conftest could access the TM_VERSIONS variable in testfiles/utils.py right ? I guess it would indeed be less confusing.

Reply:
The parametrization of the cls_dict_path fixture in test_files and test_parser now includes the version in the "files_list" parameter so that it is not longer confusing.